core/variant/variant_op.cpp

```diff
@@ -257,14 +257,22 @@ void Variant::_register_variant_operators() {
 	register_op<OperatorEvaluatorAdd<double, double, double>>(Variant::OP_ADD, Variant::FLOAT, Variant::FLOAT);
 	register_string_op(OperatorEvaluatorStringConcat, Variant::OP_ADD);
 	register_op<OperatorEvaluatorAdd<Vector2, Vector2, Vector2>>(Variant::OP_ADD, Variant::VECTOR2, Variant::VECTOR2);
+	register_op<OperatorEvaluatorAdd<Vector2, Vector2, Vector2i>>(Variant::OP_ADD, Variant::VECTOR2, Variant::VECTOR2I);
 	register_op<OperatorEvaluatorAdd<Vector2i, Vector2i, Vector2i>>(Variant::OP_ADD, Variant::VECTOR2I, Variant::VECTOR2I);
+	register_op<OperatorEvaluatorAdd<Vector2, Vector2i, Vector2>>(Variant::OP_ADD, Variant::VECTOR2I, Variant::VECTOR2);
 	register_op<OperatorEvaluatorAdd<Vector3, Vector3, Vector3>>(Variant::OP_ADD, Variant::VECTOR3, Variant::VECTOR3);
+	register_op<OperatorEvaluatorAdd<Vector3, Vector3, Vector3i>>(Variant::OP_ADD, Variant::VECTOR3, Variant::VECTOR3I);
 	register_op<OperatorEvaluatorAdd<Vector3i, Vector3i, Vector3i>>(Variant::OP_ADD, Variant::VECTOR3I, Variant::VECTOR3I);
+	register_op<OperatorEvaluatorAdd<Vector3, Vector3i, Vector3>>(Variant::OP_ADD, Variant::VECTOR3I, Variant::VECTOR3);
 	register_op<OperatorEvaluatorAdd<Vector4, Vector4, Vector4>>(Variant::OP_ADD, Variant::VECTOR4, Variant::VECTOR4);
+	register_op<OperatorEvaluatorAdd<Vector4, Vector4, Vector4i>>(Variant::OP_ADD, Variant::VECTOR4, Variant::VECTOR4I);
 	register_op<OperatorEvaluatorAdd<Vector4i, Vector4i, Vector4i>>(Variant::OP_ADD, Variant::VECTOR4I, Variant::VECTOR4I);
+	// register_op<OperatorEvaluatorAdd<Vector4, Vector4i, Vector4>>(Variant::OP_ADD, Variant::VECTOR4I, Variant::VECTOR4);
```

Comment:
Vector4i is missing some conversion that would allow this. Other vectors don't have problem, so it's likely a bug.